### DISCUSSION TITLE
Text Texture op v5 dev

### DISCUSSION THREAD

**Discussion by marioepsley** [cables-gl/cables_docs#593] [! Welcome - readme first!]
1.
Potential bug with the TextTexture op and sidebar text input area. 
When you return the lineheights change, isn't smooth.
https://dev.cables.gl/edit/AL2o7J

https://github.com/cables-gl/cables_docs/assets/36418408/ce21afb0-20a8-41bd-8d81-e78e183c3dfd

2.
if you write a big string in the sidebar say

'here we
go to check
the padding
and bits
all together'

then adjust the padding x, you'll notice things are getting squished.

<img width="446" alt="Screenshot 2023-10-19 at 11 20 08" src="https://github.com/cables-gl/cables_docs/assets/36418408/587c3dd5-6ced-4b49-a7d1-01bb7d958975">

3.
I also find the padding y top and padding y bottom more often than not greyed out.
I guess these padding should just provide more space around the edges?

4.
not sure why but it doesn't seem to reading the manual texture width and height in this comp

**Comment by marioepsley** [comment]
I made this comp for you to better illustrate some of the bugs I've found with the text texture op, hope it helps.
https://cables.gl/edit/WJYQKK

**Comment by marioepsley** [comment]
Just found something else, but the texttexture v5 doesn't seem to work nicely with the repeat op as shown here https://dev.cables.gl/edit/J9z9BM

If you disconnect the string it works, repeats "cables" 5 times. With  a string array it does something funny.
Also the left alignment doesn't work but think i've mentioned it above.

**Comment by marioepsley** [comment]
One more bug where the text texture op doesn't seem to work when iterated, only the first word shows. example here
https://dev.cables.gl/edit/pEX3WO

**Comment by pandrr** [comment]
i made a new version (v6). many problems should be solved with that.

1. fixed 
2. squish does not paggen to me
3. paddingy now should
- yes, does not work with repeat or iterator ops for now

**Comment by pandrr** [comment]
great example patch btw, thanks

**Comment by marioepsley** [comment]
Awesome, some great fixes here, the baseline fix was hands down the most important, really happy about that :). When I get a chance I'll tweak the comp to compile all the remaining issues for you. Thank you

**Comment by marioepsley** [comment]
I tweaked the comp for you https://dev.cables.gl/edit/SQSN62
I'd say the first one is the most important.
Thank you for fixing these bugs.

**Comment by pandrr** [comment]
1. should not fit over eath other, you have to multipluy the distance by 3, as you did with the font size. then the chars have the same size:

<img width="688" alt="Screenshot 2024-01-22 at 17 36 07" src="https://github.com/cables-gl/cables_docs/assets/98792/efa5fb1c-7776-4c52-8133-a343546730f7">

**Comment by pandrr** [comment]
2. fixed
3. fixed for "bottom", top was good i think?

**Comment by pandrr** [comment]
4. "line height add"means this is added to each line to the bottom. add same value to paddingY top and it should be what you want

**Comment by pandrr** [comment]
5. fixed
6. yes, bug...
7. fixed

**Comment by pandrr** [comment]
thanks again for great debugging patch! <3

**Comment by marioepsley** [comment]
Ahh thats alright, thanks for this, bit of shame when you set the size of the text texture and tweak the padding and stuff the size changes, say you just want to work in square this becomes a little fiddly but not the end of the world. certainly usable.

<img width="842" alt="Screenshot 2024-01-22 at 16 55 32" src="https://github.com/cables-gl/cables_docs/assets/36418408/cf636e23-1ee6-4fba-96b1-fda8626c4ea0">

**Comment by pandrr** [comment]
it is a pain, but i dont have a solution to this because it must be so flexible

**Reply by marioepsley** [reply]
I completely understand, the flexibility is key ultimately

**Comment by marioepsley** [comment]
assuming it's not possible to iterate the meshes https://dev.cables.gl/edit/pEX3WO, then I think we can mark this as done, I can't wait to get this moved onto the live cables. For whatever reason the patch I want to use it on takes an incredibly long time to load on dev.

**Reply by pandrr** [reply]
then it will load long on live too later. please invite me ill have a look
